### PR TITLE
Typescript time

### PR DESCRIPTION
**Pull request by alexreardon**

### CODE REVIEW

File: test/type-narrowing.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import invariant from "../src/tiny-invariant";
+import { expectType, TypeOf, TypeEqual } from "ts-expect";
```
Comment (alexreardon) [inline test/type-narrowing.spec.ts]: this lib is pretty cool!

File: src/index.d.ts [hunk 2]
```diff
@@ -1 +0,0 @@
-export default function invariant(condition: any, message?: string): void
```
Comment (flexdinesh) [inline src/index.d.ts]: @alexreardon should the return type be `void | never`?

ref: 
- https://www.typescriptlang.org/docs/handbook/basic-types.html#never
- https://stackoverflow.com/a/49435648

Comment (flexdinesh) [reply src/index.d.ts]: I see that asserts condition is the alternate for `void|never`. Can ignore this one.

File: test/type-narrowing.spec.ts (hunk 1 above)
Comment (flexdinesh) [reply test/type-narrowing.spec.ts]: this is fantastico!

**Pull request merged**